client/src/components/AddChatView.tsx

```diff
@@ -1,11 +1,76 @@
-
+import  { ChangeEvent, useState, useContext } from 'react';
+import { UserContext } from './UserProvider';
+import ProfileImageUploader from './ProfileImageUploader';
+import useAddChat from '../hooks/useAddChat';
 
 const AddChatView = () => {
-    return (
-        <div>
-            AddChatView
-        </div>
-    );
-}
-
-export default AddChatView;
\ No newline at end of file
+  const userContext = useContext(UserContext);
+
+  if (!userContext) {
+    throw new Error('AddChatView must be used within a UserProvider');
+  }
+
+  const { pfp, setPfp } = userContext;
+  const [chatName, setChatName] = useState('');
+  const [username, setUsername] = useState('');
+  const { addUserToChat, createChat, addedUsers, errorMessage } = useAddChat();
+
+  const handleChatNameChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setChatName(event.target.value);
+  };
+
+  const handleUsernameChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setUsername(event.target.value);
+  };
+
+  const handleAddUser = async () => {
+    await addUserToChat(username);
+    setUsername('');
+  };
+
+  const handleCreateChat = async () => {
+    await createChat(chatName, addedUsers, pfp);
+  };
+
+  return (
+    <div className="flex flex-col p-7">
+      <ProfileImageUploader pfp={pfp} setPfp={setPfp} />
+      <p className="mt-8 ml-2 text-left text-sm">Chat Name</p>
+      <input
```

Comment:
Te inputy sie przypadkiem nie pojawiają w settingasch też?

Można zrobić z tego też komponent, który renderuje tylko ten input - i wstawić w te trzy miejsca.

Potem jak trzeba bedize cos zmienic to tylko w jednym miejscu, a nie 3.